integration_tests/src/main/python/sort_test.py

```diff
@@ -34,6 +34,40 @@ def test_single_orderby(data_gen, order):
             lambda spark : unary_op_df(spark, data_gen).orderBy(order),
             conf = allow_negative_scale_of_decimal_conf)
 
+@pytest.mark.parametrize('shuffle_parts', [
+    pytest.param(1),
+    # This xfail would normally work but device-side asserts currently lead to memmory
+    # leaks, and they cause persitent bad state and good tests start failing
+    # pytest.param(200, marks=pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/1607"))
```

Comment:
This scares me enough that I don't want to merge it in until it is fixed. I'd rather fall back to the CPU then risk these problems.

Reply:
Disabled GPU for numPartitions > 1